src/map/status.c

```diff
@@ -1600,28 +1636,8 @@ static int status_check_skilluse(struct block_list *src, struct block_list *targ
 	if( skill_id ) {
 		if (src != NULL && (sd == NULL || sd->auto_cast_current.type != AUTOCAST_ITEM)) {
 			// Items that cast skills using 'itemskill' will not be handled by map_zone_db.
-			int i;
-
-			for(i = 0; i < map->list[src->m].zone->disabled_skills_count; i++) {
-				if( skill_id == map->list[src->m].zone->disabled_skills[i]->nameid && (map->list[src->m].zone->disabled_skills[i]->type&src->type) ) {
-					if (src->type == BL_PC) {
-#if PACKETVER >= 20080311
-						clif->skill_mapinfomessage(sd, 2);
-#else
-						clif->messagecolor_self(sd->fd, COLOR_CYAN, msg_sd(sd, 50));
-#endif
-					} else if (src->type == BL_MOB && map->list[src->m].zone->disabled_skills[i]->subtype != MZS_NONE) {
-						if( st->mode&MD_BOSS ) { /* is boss */
-							if( !( map->list[src->m].zone->disabled_skills[i]->subtype&MZS_BOSS ) )
-								break;
-						} else { /* is not boss */
-							if( map->list[src->m].zone->disabled_skills[i]->subtype&MZS_BOSS )
-								break;
-						}
-					}
-					return 0;
-				}
-			}
+			if (!status_check_skilluse_mapzone(src, st, skill_id))
```

Comment:
Please use the function pointer.
```suggestion
			if (!status->check_skilluse_mapzone(src, st, skill_id))
```